Resources/Private/CodeTemplates/Extbase/Classes/Controller/Controller.phpt

```diff
@@ -26,7 +26,7 @@ class MyController extends \TYPO3\CMS\Extbase\Mvc\Controller\ActionController
      *
      * @return void
      */
-    public function listAction()
+    public function listAction(): void
```

Comment:
That's questionable.
Actions can return `void`/`null`, `string` or any object that implements `__toString`.
See `ActionController::callActionMethod`

Reply:
I agree especially because TYPO3 11 requires to return a Response Object

Reply:
While I agree with v11 and see the parent, the current phpDoc in that generated code says it's void / nothing is returned. So void is the right declaration in my opinion.

An other approach could be to switch to the new way of doing things.

Reply:
> the current phpDoc in that generated code says it's void

Well, this should actually be corrected too then. Since we currently aim for v10 support, we can ignore v11 for the time being.

Reply:
Alright. Then I opt to remove `:void` for the actions and streamline the remaining changes in this PR. Aside from that I would provide a PR which uses the new way of doing.

Any objections?

Reply:
> Aside from that I would provide a PR which uses the new way of doing.

You mean for v11?

Reply:
Yes. It works in v10 already, doesn't it? While I understand your point that we can ignore it for the time being I don't see a reason why we should. If v10 supports it, the ext. is more future proof. I did not work with that yet. Correct me if I miss something :)

Reply:
We don't have a branch for v11 yet.

Reply:
I still don't see the reason to do it the old way but alright. Then I leave the actions as they are. So, remove `:void` from the actions. At least I would like to update the phpDoc then. So it reflects the options.

`@return ?string or none (void)`

Above is an example.

Reply:
> I still don't see the reason to do it the old way but alright. 

I guess I misunderstood you now. What would you like to do here for v10?

correct currently would be:
`@return string | object | void | null`

Reply:
> I guess I misunderstood you now. What would you like to do here for v10?

As I understood the new way of doing things - and enforced with v11 - to return the view. This is possible with v10 already afaik.

So, I think it is a good call to make use of the new way in the `phpt` file.

Reply:
No. See https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/11.0/Deprecation-92784-ExtbaseControllerActionsMustReturnResponseInterface.html

The current behavior up to v10 is deprecated now and a `ResponseInterface` must be returned, not the view.
So the new signature is totally incompatible.

Reply:
Ah I see. Thanks. Then it does not make sense indeed. So I think the acceptance criteria for this thread should be:

- [ ] Action's phpDoc is updated to `@return string | object | void | null`
- [ ] Return type declaration `:void` is removed from the actions

Reply:
No rush, I know it's OSS. This is just a bump in case you're not aware. I have applied the changes. They required a change within the test code.